File: plugins/de.ovgu.featureide.examples/featureide_examples/GPL-FH-Java/features/UndirectedWithNeighbors/GPL/Neighbor.java
package GPL;

import java.util.LinkedList;

// Vertex class

 // *************************************************************************

public class Neighbor implements EdgeIfc, NeighborIfc
{
    public  Vertex neighbor;

    // This constructor has to be present here so that the default one
    // Called on Weighted can call it, i.e. it is not longer implicit
    public Neighbor()  {
        neighbor = null;
    }

    public Neighbor( Vertex theNeighbor )
   {
        NeighborConstructor( theNeighbor );
    }

    public void setWeight(int weight) {}
    public int getWeight() { return 0; }

    public void NeighborConstructor( Vertex theNeighbor ) {
        neighbor = theNeighbor;
    }

    public void display()
    {
        System.out.print( neighbor.name + " ," );
    }

    public Vertex getStart( ) { return null; }
    public Vertex getEnd( ) { return null; }

    public Vertex getOtherVertex( Vertex vertex )
    {
        return neighbor;
    }

    public void adjustAdorns( EdgeIfc the_edge )
    {
    }
}

File: projects/GPL-FH-CSharp/features/DirectedWithNeighbors/Neighbor.cs
/*layer Directed;

import java.util.LinkedList;*/

// *************************************************************************
   
public class Neighbor : EdgeIfc, NeighborIfc
{
    public  Vertex neighbor;
        
    // This constructor has to be present here so that the default one
    // Called on Weighted can call it, i.e. it is not longer implicit    
    public Neighbor( )
    {
        neighbor = null;
    }
    
    public Neighbor( Vertex theNeighbor ) 
    {
        NeighborConstructor( theNeighbor );
    }
    
    public virtual void NeighborConstructor( Vertex theNeighbor ) 
    {
        neighbor = theNeighbor;
    }
  
    public virtual void display () 
    {
        System.Console.Out.Write( neighbor.name + " ," );
    }

    public virtual Vertex GetStart( ) 
    { 
       return null; 
    }

    public virtual Vertex GetEnd( ) 
    { 
      return neighbor; 
    }

    public virtual void setWeight( int weight ) 
    {
    }

    public virtual Vertex GetOtherVertex( Vertex vertex )
    {
        return neighbor;
    }

    public virtual void adjustAdorns( EdgeIfc the_edge )
    {
    }
}
